modules/dxp/apps/segments/segments-asah-connector-test/src/testIntegration/java/com/liferay/segments/asah/connector/internal/model/listener/test/SegmentsEntryModelListenerTest.java

```diff
@@ -0,0 +1,210 @@
+				).build(),
+				null, true, null, User.class.getName(), _serviceContext);
+
+		SegmentsExperience segmentsExperience1 =
+			_segmentsExperienceLocalService.addSegmentsExperience(
+				TestPropsValues.getUserId(), _group.getGroupId(),
+				segmentsEntry1.getSegmentsEntryId(),
+				_portal.getClassNameId(Layout.class), _layout.getPlid(),
+				Collections.singletonMap(
+					LocaleUtil.getSiteDefault(), "Segment experience 1"),
+				true, new UnicodeProperties(true), _serviceContext);
+
+		SegmentsExperiment segmentsExperiment =
+			_segmentsExperimentLocalService.addSegmentsExperiment(
+				segmentsExperience1.getSegmentsExperienceId(),
+				_portal.getClassNameId(Layout.class), _layout.getPlid(),
+				"AB test", "A/B test description",
+				SegmentsExperimentConstants.Goal.BOUNCE_RATE.getLabel(),
+				StringPool.BLANK, _serviceContext);
+
+		Object asahSegmentsExperimentProcessor =
+			ReflectionTestUtil.getFieldValue(
+				_modelListener, "_asahSegmentsExperimentProcessor");
+
+		Object asahFaroBackendClient = ReflectionTestUtil.getFieldValue(
+			asahSegmentsExperimentProcessor, "_asahFaroBackendClient");
+
+		ReflectionTestUtil.setFieldValue(
+			asahFaroBackendClient, "_http",
+			MockHttpUtil.geHttp(
+				Collections.singletonMap(
+					"/api/1.0/experiments/" +
+						segmentsExperiment.getSegmentsExperienceId(),
+					() -> "{}")));
+
+		SegmentsExperience segmentsExperience2 =
+			_segmentsExperienceLocalService.addSegmentsExperience(
+				TestPropsValues.getUserId(), _group.getGroupId(), 0,
+				_portal.getClassNameId(Layout.class), _layout.getPlid(),
+				Collections.singletonMap(
+					LocaleUtil.getSiteDefault(), "Variant 1"),
+				false, new UnicodeProperties(true), _serviceContext);
+
+		_segmentsExperimentRelService.addSegmentsExperimentRel(
+			segmentsExperiment.getSegmentsExperimentId(),
+			segmentsExperience2.getSegmentsExperienceId(), _serviceContext);
+
+		try (CompanyConfigurationTemporarySwapper
+				companyConfigurationTemporarySwapper =
+					new CompanyConfigurationTemporarySwapper(
+						TestPropsValues.getCompanyId(),
+						AnalyticsConfiguration.class.getName(),
+						HashMapDictionaryBuilder.<String, Object>put(
+							"liferayAnalyticsDataSourceId",
+							RandomTestUtil.randomLong()
+						).put(
+							"liferayAnalyticsEnableAllGroupIds", true
+						).put(
+							"liferayAnalyticsFaroBackendSecuritySignature",
+							RandomTestUtil.randomString()
+						).put(
+							"liferayAnalyticsFaroBackendURL",
+							RandomTestUtil.randomString()
+						).build(),
+						SettingsFactoryUtil.getSettingsFactory())) {
+
+			String name = "Segments entry modified";
+
+			segmentsEntry1.setNameMap(
+				Collections.singletonMap(LocaleUtil.getSiteDefault(), name));
+
+			_segmentsEntryLocalService.updateSegmentsEntry(segmentsEntry1);
+
+			SegmentsEntry segmentsEntry2 =
+				_segmentsEntryLocalService.getSegmentsEntry(
+					segmentsEntry1.getSegmentsEntryId());
+
+			Assert.assertEquals(
```

Comment:
Here you are not asserting the update call to AC, so maybe this test should be translated into a unit test or not tested at all.

Reply:
This class has been deleted and I added some unit tests of AsahFaroBackendClientImplTest class.